assets/app/view/game/discard_trains.rb

```diff
@@ -35,7 +35,9 @@ def render
             h(:div, trains),
           ])
         end
-
+        overflow << h(ScrapTrains, corporation: @current_entity) if @game.round.actions_for(
+            @game.round.active_step&.current_entity
```

Comment:
active_step is assigned to step above. Can you use that instead of triggering active_step again?

Reply:
I've updated this block